jaeger-core/src/main/java/com/uber/jaeger/propagation/B3TextMapCodec.java

```diff
@@ -47,44 +49,52 @@
 
   @Override
   public void inject(SpanContext spanContext, TextMap carrier) {
-    carrier.put(TRACE_ID_NAME, HexCodec.toLowerHex(spanContext.getTraceId()));
-    if (spanContext.getParentId() != 0L) { // Conventionally, parent id == 0 means the root span
-      carrier.put(PARENT_SPAN_ID_NAME, HexCodec.toLowerHex(spanContext.getParentId()));
-    }
-    carrier.put(SPAN_ID_NAME, HexCodec.toLowerHex(spanContext.getSpanId()));
-    carrier.put(SAMPLED_NAME, spanContext.isSampled() ? "1" : "0");
-    if (spanContext.isDebug()) {
-      carrier.put(FLAGS_NAME, "1");
+    try {
+      carrier.put(TRACE_ID_NAME, HexCodec.toLowerHex(spanContext.getTraceId()));
+      if (spanContext.getParentId() != 0L) { // Conventionally, parent id == 0 means the root span
+        carrier.put(PARENT_SPAN_ID_NAME, HexCodec.toLowerHex(spanContext.getParentId()));
+      }
+      carrier.put(SPAN_ID_NAME, HexCodec.toLowerHex(spanContext.getSpanId()));
+      carrier.put(SAMPLED_NAME, spanContext.isSampled() ? "1" : "0");
+      if (spanContext.isDebug()) {
+        carrier.put(FLAGS_NAME, "1");
+      }
+    } catch (RuntimeException ex) {
+      log.error("Error when injecting SpanContext into carrier. Handling gracefully.", ex);
```

Comment:
this can fill logs. I'd make it debug level

Reply:
to put in perspective. once there was a proxy which shipped with an encoding but (parentId = literally '-') it wasn't fixed for months. that would be a lot of error log statements!

Reply:
* it would have been probably logged anyway if it had had bubble up the stack
* there is no other way to notify the developer that something doesn't work well. Especially in case of Span injection which will be most probably caused by developers error.
* In cases like the one you've mentioned it can be always muted on a logger level with configuration.